### PR TITLE
Add more CloudEvents around the management of ScaledObjects resources

### PR DESCRIPTION
**Pull request by SpiritZhou**
Add more CloudEvents around the management of ScaledObjects resources

### Checklist

- [X] Commits are signed with Developer Certificate of Origin (DCO)

Relates to https://github.com/kedacore/keda/issues/3522
Relates to https://github.com/kedacore/keda/pull/5953

### CODE REVIEW

File: content/docs/2.15/operate/events.md [hunk 1]
```diff
@@ -141,4 +156,5 @@ eventSubscription: #Optional. Submit included/excluded event types will filter e
 | Event Type                                | Scenario Description                                                                                                                 | 
 | ------------------------------------- |  --------------------------------------------------------------------------------------------------------------------------- | 
 | `keda.scaledobject.ready.v1`                   | On the first time a ScaledObject is ready, or if the previous ready condition status of the object was `Unknown` or `False` |  
-| `keda.scaledobject.failed.v1`             | If the check validation for a ScaledObject fails |      
\ No newline at end of file
+| `keda.scaledobject.failed.v1`             | If the check validation for a ScaledObject fails |         
+| `keda.scaledobject.removed.v1`             | a ScaledObject is deleted |
```
Comment (tomkerkhove) [inline content/docs/2.15/operate/events.md:160]:
```suggestion
| `keda.scaledobject.removed.v1`             | When a ScaledObject is deleted |   
```

**Review by tomkerkhove — APPROVED** [review]

**Comment by tomkerkhove** [comment]
@SpiritZhou Can you fix merge conflicts please?

**Comment by SpiritZhou** [comment]
> @SpiritZhou Can you fix merge conflicts please?

Fixed.

**Comment by tomkerkhove** [comment]
Thank you!

**Pull request merged**